pyvista/core/grid.py

```diff
@@ -575,3 +575,21 @@ def cast_to_structured_grid(self):
         alg.SetInputData(self)
         alg.Update()
         return _get_output(alg)
+
+
+    def cast_to_rectilinear_grid(self):
+        """Cast this unifrom grid to a :class:`pyvista.RectilinearGrid`"""
+        def gen_coords(i):
+            coords = np.cumsum(np.insert(np.full(self.dimensions[i] - 1,
+                                                 self.spacing[i]), 0, 0)
+                               ) + self.origin[i]
+            return coords
+        xcoords = gen_coords(0)
+        ycoords = gen_coords(1)
+        zcoords = gen_coords(2)
+        grid = pyvista.RectilinearGrid(xcoords, ycoords, zcoords)
+        grid.point_arrays.update(self.point_arrays)
```

Comment:
Do we want to copy these dictionaries so the cast object does not reference the old object?

Reply:
Looks like they are actually deep copied